app/scripts/modules/core/src/presentation/hooks/useDataSource.hook.ts

```diff
@@ -0,0 +1,32 @@
+import { useObservableValue } from './useObservableValue.hook';
+import { useLatestCallback } from './useLatestCallback.hook';
+
+import { ApplicationDataSource } from '../../application';
+
+export interface IDataSourceResult<T> {
+  data: T;
+  refresh: () => void;
+}
+
+/**
+ * A react hook that returns the current value an ApplicationDataSource
+ * and triggers a re-render when a new value is set. Also provides a function
+ * for refreshing the data source manually.
+ *
+ * @param dataSource the data source to subscribe to
+ * @returns IDataSourceResult<T>
+ */
+export const useDataSource = <T>(dataSource: ApplicationDataSource<T>): IDataSourceResult<T> => {
+  const data = useObservableValue(dataSource.data$, dataSource.data);
```

Comment:
any reason you didn't use `dataSource.data$.value`?

Reply:
yeah, that'd work too. I haven't really worked with behavior subjects much so didn't even come to mind. will update.

Reply:
done!

Reply:
It appears that `dataSource.data$.value` is initially undefined whereas `dataSource.data` contains the `defaultData` 🤷‍♂

Reply:
Looks like it's because
```
public data$ = new BehaviorSubject<T>(this.data);
```
runs before
```
this.data = this.data || this.defaultData;
```
in the `constructor`